### PR TITLE
Extract RuboCop FactoryBot from RuboCop RSpec

### PR DESCRIPTION
**Pull request by ydah**
This PR is extract RuboCop FactoryBot from RuboCop RSpec.

______________________________________________________________________

Before submitting the PR make sure the following are checked:

- [x] Feature branch is up-to-date with `master` (if not - rebase it).
- [x] Squashed related commits together.
- [x] Added tests.
- [x] Updated documentation.
- [x] Added an entry to the `CHANGELOG.md` if the new code introduces user-observable changes.
- [x] The build (`bundle exec rake`) passes (be sure to run this locally, since it may produce updated documentation that you will need to commit).

### CODE REVIEW

File: rubocop-factory_bot.gemspec [hunk 1]
```diff
@@ -1,25 +1,20 @@
...
+  spec.name = 'rubocop-factory_bot'
+  spec.summary = 'Code style checking for FactoryBot files'
   spec.description = <<-DESCRIPTION
-    Code style checking for RSpec files.
+    Code style checking for FactoryBot files.
     A plugin for the RuboCop code style enforcing & linting tool.
   DESCRIPTION
-  spec.homepage = 'https://github.com/rubocop/rubocop-rspec'
+  spec.homepage = 'https://github.com/rubocop/rubocop-factory_bot'
   spec.authors = ['John Backus', 'Ian MacLeod', 'Nils Gemeinhardt']
```
Comment (ydah) [inline rubocop-factory_bot.gemspec:14]: memo) The author has not changed from RuboCop RSpec, but who is appropriate?

Comment (pirj) [reply rubocop-factory_bot.gemspec:14]: I suggest blaming the lib/cops folder and seeing who the committers were.
For capybara cops that was almost exclusively you, @ydah.

Side note: @r7kamura you seem to have in-depth knowledge of FB, would you be actively contributing to this repo?

**Comment by pirj** [comment]
My only concern regarding this PR is that `master` would still have traces of old file versions from `rubocop-rspec`.
I'd rather start `master` from the point where `factory_bot-only` is.

File: rubocop-factory_bot.gemspec (hunk 1 above)
Comment (r7kamura) [reply rubocop-factory_bot.gemspec:14]: Thank you for your efforts on this gem split 👍 
I don't have the confidence at this time to be involved in the ongoing maintenance of this gem, but I do have a few new cop ideas I would like to add.

Comment (ydah) [reply rubocop-factory_bot.gemspec:14]: I was looking at the results of git blame and it did not seem that many FactoryBot department were committed by any particular contributor.

### Number of PR
```
pirj 4
Darhazer 4
abrom 4
composerinteralia 3
ydah 3
r7kamura 2
bquorning 2
jfragoulis 2
```

### Author of the cop.
```
FactoryBot/FactoryNameStyle: ydah
FactoryBot/ConsistentParenthesesStyle: Liberatys
FactoryBot/SyntaxMethods: leoarnold
FactoryBot/FactoryClassName: jfragoulis
FactoryBot/AttributeDefinedStatically: composerinteralia, seanpdoyle
FactoryBot/CreateList: Darhazer
FactoryBot/StaticAttributeDefinedDynamically: abrom
```

**Comment by ydah** [comment]
> My only concern regarding this PR is that `master` would still have traces of old file versions from `rubocop-rspec`. I'd rather start `master` from the point where `factory_bot-only` is.

Indeed it is. I will pushing commits directly to master instead of merging this PR.

File: rubocop-factory_bot.gemspec [hunk 2]
```diff
@@ -1,25 +1,20 @@
...
   DESCRIPTION
-  spec.homepage = 'https://github.com/rubocop/rubocop-rspec'
-  spec.authors = ['John Backus', 'Ian MacLeod', 'Nils Gemeinhardt']
-  spec.email = [
-    'johncbackus@gmail.com',
-    'ian@nevir.net',
-    'git@nilsgemeinhardt.de'
-  ]
+  spec.homepage = 'https://github.com/rubocop/rubocop-factory_bot'
+  spec.authors = ['Phil Pirozhkov', 'Maxim Krizhanovsky', 'Andrew Bromwich']
```
[the remark this answers is not in the document]
Comment (pirj) [reply rubocop-factory_bot.gemspec:14]: I'd keep the first two 😄

**Review by bquorning — COMMENTED** [review]
I think you can delete the `rspec4` job from .github/workflows/main.yml, since it’s more rubocop-rspec specific.

**Review by bquorning — APPROVED** [review]
Besides one comment, the changes look fine to me.

**Comment by ydah** [comment]
> I think you can delete the `rspec4` job from .github/workflows/main.yml, since it’s more rubocop-rspec specific.

Indeed it is. I updated this PR. Thank you!
If all is well, push to master and start this repository.

**Review by bquorning — APPROVED** [review]

**Comment by pirj** [comment]
What concerns me slightly is that we kept unrelated commits from rubocop-rspec.
CFCB8B00-710C-4D59-825C-BBB6582C98A6
is this because they touched commonly present files like the changelog and default conf?
I can’t recall if it was the same for rubocop-capybara.
My apologies for such a purist question.

**Comment by pirj** [comment]
Please disregard my previous comment. Old unrelated commits are indeed kept there, too https://github.com/rubocop/rubocop-capybara/commit/b02eb1416e9f1848deb110c47aef8c9f0a77e3bb

**Comment by pirj** [comment]
Thanks a lot for this enourmous effort, @ydah !

**Comment by bquorning** [comment]
Oh, I too was hoping to get a “clean” commit history. But yeah, I guess it would be very labor intensive to filter commits touching the Changelog file.

**Pull request merged**